card/src/main/java/com/adyen/checkout/card/ui/AddressFormInput.kt

```diff
@@ -396,29 +376,38 @@ class AddressFormInput @JvmOverloads constructor(
     fun updateAddressHint(isOptional: Boolean) {
         val spec = AddressSpecification.fromString(component.inputData.address.country)
 
-        val countryStyleResId = if (isOptional) spec.country.optionalStyleResId else spec.country.styleResId
-        textInputLayoutCountry?.setLocalizedHintFromStyle(countryStyleResId, localizedContext)
-
-        val streetStyleResId = if (isOptional) spec.street.optionalStyleResId else spec.street.styleResId
-        textInputLayoutStreet?.setLocalizedHintFromStyle(streetStyleResId, localizedContext)
+        val streetStyleResId = spec.street.getStyleResId(isOptional)
+        streetStyleResId?.let {
+            textInputLayoutStreet?.setLocalizedHintFromStyle(it, localizedContext)
+        }
 
-        val houseNumberStyleResId = if (isOptional) spec.houseNumber.optionalStyleResId else spec.houseNumber.styleResId
-        textInputLayoutHouseNumber?.setLocalizedHintFromStyle(houseNumberStyleResId, localizedContext)
+        val houseNumberStyleResId = spec.houseNumber.getStyleResId(isOptional)
```

Comment:
Minor: you could directly use `spec.houseNumber.getStyleResId(isOptional)?.let{` to reduce one line

Reply:
I like it better this way because it separates the two operations we're doing, so slightly better readability.

Reply:
👍